piglow/examples/strobe.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/goiot/devices/piglow"
+	"golang.org/x/exp/io/i2c"
+)
+
+func main() {
+	p, err := piglow.Open(&i2c.Devfs{Dev: "/dev/i2c-1", Addr: piglow.Address})
+	if err != nil {
+		panic(err)
+	}
+
+	// catch signals and terminate the app
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	if err := p.Setup(); err != nil {
+		panic(err)
+	}
+
+	for {
+		select {
+		case <-sigc:
+			p.Shutdown()
+			p.Close()
```

Comment:
The descriptor will be automatically closed on exit, please don't do this. It just makes the sample impossible to read/impossible to look for the fundamental reference parts.